mitodl/micromasters
# Clean up frontend code touching coupons

#### What are the relevant tickets?
Fixes #3261 

#### What's this PR do?
Code cleanup to prepare for #3345 

#### How should this be manually tested?
In general nothing should change. First, check out master then run `./scripts/test/run_snapshot_dashboard_states.sh --match coupon` then rename `dashboard_states_output` to `before_refactor`.

Next, check out this branch and run the same command. Rename `dashboard_states_output` to `after_refactor`. Compare the screenshots and make sure nothing changed specific to the coupons. (The names and certain grade percents will change.)
